sdk/servicebus/service-bus/samples/v7/javascript/session.js

```diff
@@ -79,22 +81,44 @@ async function sendMessage(sbClient, scientist, sessionId) {
 
 async function receiveMessages(sbClient, sessionId) {
   // If receiving from a subscription you can use the acceptSession(topic, subscription, sessionId) overload
-  const receiver = await sbClient.acceptSession(queueName, sessionId);
-
-  const processMessage = async (message) => {
-    console.log(`Received: ${message.sessionId} - ${message.body} `);
-  };
-  const processError = async (args) => {
-    console.log(`>>>>> Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-  receiver.subscribe({
-    processMessage,
-    processError
-  });
-
-  await delay(5000);
-
-  await receiver.close();
+  let endDate;
+
+  while (true) {
+    console.log(`Creating session receiver for session '${sessionId}'`);
+    const receiver = await sbClient.acceptSession(queueName, sessionId);
+
+    const subscribePromise = new Promise((_, reject) => {
+      const processMessage = async (message) => {
+        console.log(`Received: ${message.sessionId} - ${message.body} `);
+      };
+      const processError = async (args) => {
+        console.log(`>>>>> Error from error source ${args.errorSource} occurred: `, args.error);
+        reject(args.error);
+      };
+
+      receiver.subscribe({
+        processMessage,
+        processError
+      });
+    });
+
+    const now = Date.now();
+    endDate = endDate ?? now + 20000;
+    let remainingTime = endDate - now;
+
+    console.log(`Waiting for ${remainingTime} milliseconds for messages to arrive.`);
+
+    try {
+      await Promise.race([subscribePromise, delay(remainingTime)]);
+
+      // wait time has expired, we can stop listening.
+      console.log(`Time has expired, closing receiver for session '${sessionId}'`);
```

Comment:
not closing the receiver when the time expires?

Reply:
I'll add it in. There's an overall ServiceBusClient.close() that technically takes care of it but that's not best practice :)

Reply:
Fixed.